gradle-plugins/model-compiler/src/main/java/io/spine/gradle/compiler/lookup/valrule/ValidationRulesLookupPlugin.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.gradle.compiler.lookup.valrule;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.DescriptorProtos.FileDescriptorProto;
+import io.spine.gradle.SpinePlugin;
+import io.spine.gradle.compiler.util.DescriptorSetUtil.IsNotGoogleProto;
+import io.spine.gradle.compiler.util.PropertiesWriter;
+import org.gradle.api.Action;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.slf4j.Logger;
+
+import java.util.Collection;
+import java.util.Map;
+
+import static com.google.common.collect.Maps.newHashMap;
+import static io.spine.gradle.TaskName.FIND_TEST_VALIDATION_RULES;
+import static io.spine.gradle.TaskName.FIND_VALIDATION_RULES;
+import static io.spine.gradle.TaskName.GENERATE_PROTO;
+import static io.spine.gradle.TaskName.GENERATE_TEST_PROTO;
+import static io.spine.gradle.TaskName.PROCESS_RESOURCES;
+import static io.spine.gradle.TaskName.PROCESS_TEST_RESOURCES;
+import static io.spine.gradle.compiler.Extension.getMainDescriptorSetPath;
+import static io.spine.gradle.compiler.Extension.getMainTargetGenResourcesDir;
+import static io.spine.gradle.compiler.Extension.getTestDescriptorSetPath;
+import static io.spine.gradle.compiler.Extension.getTestTargetGenResourcesDir;
+import static io.spine.gradle.compiler.util.DescriptorSetUtil.getProtoFileDescriptors;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Finds Protobuf definitions of validation rules and creates a {@code .properties} file,
+ * which contains entries like:
+ *
+ * <p>{@code VALIDATION_RULE_TYPE=MESSAGE_TYPE_FOR_VALIDATION_RULE.FIELD_NAME}
```

Comment:
This will probably read better: `{@code foo.bar.ValidationRule=alpha.beta.TargetMessage.name_of_field_to_validate}`.